File: MeatGrinder.Web/Controllers/CookieService.fs
namespace MeatGrinder.Web.Services
open System
open System.Web
[<Sealed>]
[<AbstractClass>]
type CookieService =
    static member GetCookie (context:HttpContextBase) cookieName =
        let httpCookie = context.Request.Cookies.["MeatGrinder"]
        if httpCookie<>null then Some(httpCookie.[cookieName]) else None
    static member SetCookie (response:HttpResponseBase, cookieName, cookieDuration:int, cookieValue) =
        let cookie = new HttpCookie("MeatGrinder")
        cookie.[cookieName] <- cookieValue
        cookie.Expires <- DateTime.Now.AddDays (float(cookieDuration))
        response.Cookies.Add(cookie)
    static member SetCookie (response:HttpResponseBase, cookieName, cookieDuration) =
        CookieService.SetCookie(response, cookieName, cookieDuration, "0")
    static member DeleteCookie response cookieName =
        CookieService.SetCookie(response, cookieName, -1)
    static member GetUserId () =
        let context = System.Web.HttpContextWrapper System.Web.HttpContext.Current
        let cookie = CookieService.GetCookie context "UserId"
        if cookie.IsSome && cookie.Value<> null then Some(Int32.Parse(cookie.Value)) else None

File: MeatGrinder/Services/CookieService.cs
﻿using System;
using System.Web;

namespace MeatGrinder.Services
{
    public static class CookieService
    {
        public static string GetCookie(HttpContext context, string cookieName)
        {
            var httpCookie = HttpContext.Current.Request.Cookies["meatgrinder"];
            if (httpCookie != null) 
                return httpCookie[cookieName];

            return null;
        }
        public static void SetCookie(HttpResponseBase response, string cookieName, int cookieDuration, string cookieValue = "0")
        {
            var cookie = new HttpCookie("MeatGrinder");

            cookie[cookieName] = cookieValue;
            cookie.Expires = DateTime.Now.AddDays(cookieDuration);
            response.Cookies.Add(cookie);
        }
        public static void DeleteCookie(HttpResponseBase response, string cookieName)
        {
            SetCookie(response, cookieName, -1);
        }
        public static int GetUserID()
        {
            int userID = int.Parse(CookieService.GetCookie(System.Web.HttpContext.Current, "UserID"));
            return userID;
        }

    }
}